test/internal/Compose.js

```diff
@@ -23,11 +25,11 @@ module.exports = function Compose(F) {
     };
 
     ComposeFG.prototype[FL.map] = function(f) {
-      return ComposeFG(Z.map(function(y) { return Z.map(f, y); }, this.value));
+      return ComposeFG(Z.map(S.map(f), this.value));
     };
 
     ComposeFG.prototype[FL.ap] = function(other) {
-      return ComposeFG(Z.ap(Z.map(function(u) { return function(y) { return Z.ap(u, y); }; }, other.value), this.value));
+      return ComposeFG(Z.ap(Z.map(S.ap, other.value), this.value));
```

Comment:
I think that would a good approach, as when we move this laws in a separate module, we might not want it to be depend on sanctuary, because it has curried functions .